### ISSUE TITLE
RSpec/MatchArray: Incorrect correction applied, causes failure in inspection

### ISSUE THREAD

**Issue by CKolkey** [rubocop/rubocop#11684]
When running `bundle exec rubocop -a` I encountered an issue with the MatchArray cop.

The following correction is offered:
```
spec.rb:32:32: C: [Correctable] RSpec/MatchArray: Prefer contain_exactly when matching an array literal.
        expect(action.call).to match_array([])
                               ^^^^^^^^^^^^^^^
```

And this is what was applied:
```diff
       it "returns an empty array" do
-        expect(action.call).to match_array([])
+        expect(action.call).to match_array
       end
```

Which results in:
```
undefined method `array_type?' for nil:NilClass

          return unless node.first_argument.array_type?
                                           ^^^^^^^^^^^^
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.19.0/lib/rubocop/cop/rspec/match_array.rb:27:in `on_send'
```

<details>
<summary>Backtrace</summary>
An error occurred while RSpec/MatchArray cop was inspecting /Users/u0158204/code/karnov/pro/spec/services/content_references_migration/dk/migrate_content_reference_service_spec.rb:167:51.
undefined method `array_type?' for nil:NilClass

          return unless node.first_argument.array_type?
                                           ^^^^^^^^^^^^
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.19.0/lib/rubocop/cop/rspec/match_array.rb:27:in `on_send'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:143:in `public_send'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:143:in `block (2 levels) in trigger_restricted_cops'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:142:in `block in trigger_restricted_cops'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:141:in `each'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:141:in `trigger_restricted_cops'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:156:in `each'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:156:in `on_send'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_send'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:156:in `investigate_partial'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:91:in `investigate'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:345:in `block in inspect_file'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:344:in `each'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:344:in `flat_map'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:344:in `inspect_file'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:314:in `loop'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:283:in `do_inspection_loop'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:164:in `block in file_offenses'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:189:in `file_offense_cache'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:163:in `file_offenses'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:154:in `process_file'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:134:in `each'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:134:in `reduce'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:134:in `each_inspected_file'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:120:in `inspect_files'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:73:in `run'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:118:in `run_command'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:125:in `execute_runners'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:51:in `block in run'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:43:in `run'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/exe/rubocop:19:in `block in <top (required)>'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/benchmark-0.2.1/lib/benchmark.rb:311:in `realtime'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/exe/rubocop:19:in `<top (required)>'
/Users/u0158204/.asdf/installs/ruby/3.1.3/bin/rubocop:25:in `load'
/Users/u0158204/.asdf/installs/ruby/3.1.3/bin/rubocop:25:in `<top (required)>'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli/exec.rb:58:in `load'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli/exec.rb:58:in `kernel_load'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli/exec.rb:23:in `run'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli.rb:491:in `exec'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli.rb:34:in `dispatch'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli.rb:28:in `start'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/exe/bundle:45:in `block in <top (required)>'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
/Users/u0158204/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/exe/bundle:33:in `<top (required)>'
/Users/u0158204/.asdf/installs/ruby/3.1.3/bin/bundle:25:in `load'
/Users/u0158204/.asdf/installs/ruby/3.1.3/bin/bundle:25:in `<main>'
</details>

## Expected behavior
Not to fail.

## Actual behavior
Failed.

## Steps to reproduce the problem
`spec/test_spec.rb`:
```ruby
# frozen_string_literal: true

RSpec.describe "test" do
  let(:value) { [] }

  it "is empty" do
    expect(value).to match_array([])
  end
end
```

```
> bundle exec rubocop spec/test_spec.rb -a -d
For .: configuration from ./.rubocop.yml
Inheriting configuration from ./.rubocop_config.yml
configuration from ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-performance-1.16.0/config/default.yml
configuration from ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-performance-1.16.0/config/default.yml
Default configuration from ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/config/default.yml
configuration from ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-rails-2.18.0/config/default.yml
configuration from ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-rails-2.18.0/config/default.yml
configuration from ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-capybara-2.17.1/config/default.yml
configuration from ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-capybara-2.17.1/lib/../config/default.yml
configuration from ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.19.0/config/default.yml
configuration from ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.19.0/config/default.yml
Inheriting configuration from ./.rubocop_todo.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning ./spec/test_spec.rb
Loading cache from .cache/rubocop_cache/886750e31dd700a20380b6a6ed8938201faebd01/8123c7090d4812cc10a91c1343176ff6e02c422b/326a6d9070747832f9b15781c4dbb2d4920af35a
An error occurred while RSpec/MatchArray cop was inspecting ./spec/test_spec.rb:7:21.
undefined method `array_type?' for nil:NilClass

          return unless node.first_argument.array_type?
                                           ^^^^^^^^^^^^
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.19.0/lib/rubocop/cop/rspec/match_array.rb:27:in `on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:143:in `public_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:143:in `block (2 levels) in trigger_restricted_cops'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:142:in `block in trigger_restricted_cops'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:141:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:141:in `trigger_restricted_cops'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:156:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:156:in `on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:20:in `walk'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:87:in `investigate'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:156:in `investigate_partial'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:91:in `investigate'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:345:in `block in inspect_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:344:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:344:in `flat_map'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:344:in `inspect_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:314:in `loop'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:283:in `do_inspection_loop'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:164:in `block in file_offenses'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:189:in `file_offense_cache'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:163:in `file_offenses'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:154:in `process_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:134:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:134:in `reduce'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:134:in `each_inspected_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:120:in `inspect_files'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:73:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command.rb:11:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/environment.rb:18:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:118:in `run_command'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:125:in `execute_runners'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:51:in `block in run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:43:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/exe/rubocop:19:in `block in <top (required)>'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/benchmark-0.2.1/lib/benchmark.rb:311:in `realtime'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/exe/rubocop:19:in `<top (required)>'
ruby/3.1.3/bin/rubocop:25:in `load'
ruby/3.1.3/bin/rubocop:25:in `<top (required)>'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli/exec.rb:58:in `load'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli/exec.rb:58:in `kernel_load'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli/exec.rb:23:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli.rb:491:in `exec'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli.rb:34:in `dispatch'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli.rb:28:in `start'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/exe/bundle:45:in `block in <top (required)>'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/exe/bundle:33:in `<top (required)>'
ruby/3.1.3/bin/bundle:25:in `load'
ruby/3.1.3/bin/bundle:25:in `<main>'
An error occurred while RSpec/MatchArray cop was inspecting ./spec/test_spec.rb:7:21.
undefined method `array_type?' for nil:NilClass

          return unless node.first_argument.array_type?
                                           ^^^^^^^^^^^^
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.19.0/lib/rubocop/cop/rspec/match_array.rb:27:in `on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:143:in `public_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:143:in `block (2 levels) in trigger_restricted_cops'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:142:in `block in trigger_restricted_cops'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:141:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:141:in `trigger_restricted_cops'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:156:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:156:in `on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_send'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.27.0/lib/rubocop/ast/traversal.rb:20:in `walk'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/commissioner.rb:87:in `investigate'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:156:in `investigate_partial'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:91:in `investigate'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:345:in `block in inspect_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:344:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:344:in `flat_map'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:344:in `inspect_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:314:in `loop'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:283:in `do_inspection_loop'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:164:in `block in file_offenses'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:189:in `file_offense_cache'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:163:in `file_offenses'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:154:in `process_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:134:in `each'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:134:in `reduce'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:134:in `each_inspected_file'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:120:in `inspect_files'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/runner.rb:73:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/command.rb:11:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli/environment.rb:18:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:118:in `run_command'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:125:in `execute_runners'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:51:in `block in run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/lib/rubocop/cli.rb:43:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/exe/rubocop:19:in `block in <top (required)>'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/benchmark-0.2.1/lib/benchmark.rb:311:in `realtime'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.48.0/exe/rubocop:19:in `<top (required)>'
ruby/3.1.3/bin/rubocop:25:in `load'
ruby/3.1.3/bin/rubocop:25:in `<top (required)>'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli/exec.rb:58:in `load'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli/exec.rb:58:in `kernel_load'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli/exec.rb:23:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli.rb:491:in `exec'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli.rb:34:in `dispatch'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/cli.rb:28:in `start'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/exe/bundle:45:in `block in <top (required)>'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
ruby/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.4.3/exe/bundle:33:in `<top (required)>'
ruby/3.1.3/bin/bundle:25:in `load'
ruby/3.1.3/bin/bundle:25:in `<main>'
C

Offenses:

spec/test_spec.rb:7:22: C: [Corrected] RSpec/ContainExactly: Prefer match_array when matching array values.
    expect(value).to contain_exactly()
                     ^^^^^^^^^^^^^^^^^
spec/test_spec.rb:7:22: C: [Corrected] RSpec/MatchArray: Prefer contain_exactly when matching an array literal.
    expect(value).to match_array([])
                     ^^^^^^^^^^^^^^^
spec/test_spec.rb:7:33: C: [Corrected] Style/MethodCallWithoutArgsParentheses: Do not use parentheses for method calls with no arguments.
    expect(value).to match_array()
                                ^^
spec/test_spec.rb:7:37: C: [Corrected] Style/MethodCallWithoutArgsParentheses: Do not use parentheses for method calls with no arguments.
    expect(value).to contain_exactly()
                                    ^^

1 file inspected, 4 offenses detected, 4 offenses corrected

2 errors occurred:
An error occurred while RSpec/MatchArray cop was inspecting ./spec/test_spec.rb:7:21.
An error occurred while RSpec/MatchArray cop was inspecting ./spec/test_spec.rb:7:21.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.48.0 (using Parser 3.2.1.1, rubocop-ast 1.27.0, running on ruby 3.1.3) [x86_64-darwin21]
Finished in 0.783444999717176 seconds
```

## RuboCop version

```
1.48.0 (using Parser 3.2.1.1, rubocop-ast 1.27.0, running on ruby 3.1.3) [x86_64-darwin21]
  - rubocop-performance 1.16.0
  - rubocop-rails 2.18.0
  - rubocop-rspec 2.19.0
```
